platform/golden/bullseye.pkr.hcl

```diff
@@ -76,37 +90,49 @@ source "virtualbox-iso" "base-debian-amd64" {
   skip_nat_mapping     = false
   iso_checksum         = "${var.iso_checksum_type}:${var.iso_checksum}"
   iso_url              = "${var.iso_url}"
-  iso_interface        = "virtio"
+  iso_interface        = "sata"
   nic_type             = "virtio"
   hard_drive_interface = "virtio"
   rtc_time_base        = "UTC"
   shutdown_command     = "sudo -S shutdown -P now"
   ssh_username         = "packer"
   ssh_password         = "${var.ssh_password}"
   ssh_wait_timeout     = "30m"
+  firmware             = "efi"
+  # don't remove the built VM from VirtualBox after export
+  keep_registered      = "false"
+  # don't export the built VM
+  skip_export          = "false"
   vboxmanage = [
     # enable recording video of install process, for debug and build record
     [ "modifyvm", "{{.Name}}", "--recording", "on" ],
-    # enable proxy access across NAT network interface.
     [ "modifyvm", "{{.Name}}", "--nat-localhostreachable1", "on" ],
+    [ "modifyvm", "{{.Name}}", "--firmware", "EFI" ],
+    ["modifyvm", "{{.Name}}", "--vram", "16"]
   ]
 }
 
-# Build a golden image by connecting a compute source with a provisioner
-
+## Build a golden image by connecting a compute source with a provisioner
+#
 build {
   sources = ["source.virtualbox-iso.base-debian-amd64"]
-
-# Use Ansible (needs to be on the "packer" host) to provision the instance
   provisioner "ansible" {
     playbook_file = "./provisioners/01_update_packer_user/packer.yml"
     ansible_env_vars = [
       "ANSIBLE_HOST_KEY_CHECKING=False",
       "ANSIBLE_SSH_ARGS='-oForwardAgent=yes -oHostKeyAlgorithms=+ssh-rsa -oPubkeyAcceptedKeyTypes=ssh-rsa'"
     ]
-    # Add scp-extra-args -O to address inconsistencies with SFTP protocol
     extra_arguments = [ "--scp-extra-args", "'-O'" ]
-    # Set username to install user 'packer'
-    user="packer"
+    user = "packer"
```

Comment:
Line `103`-`108` dont work for me. I have a small patch which I htink fixes it up.

Would you consider integrating this diff into your work?

```
Paste diff
```

Reply:
I think the diff is missing.  Not sure what you mean "doesn't work" as in you don't like it, or as in it throws an error?

Reply:
Patch from below didn't work either, but I did go through and fix up the script.  The latest push should have the minimum set that works for me at least against VB 7.0.6.